js/src/shared/util/wallets.js

```diff
@@ -0,0 +1,405 @@
+          return bool;
+        });
+    }
+
+    return _cachedWalletLookup[address];
+  }
+
+  static fetchRequire (walletContract) {
+    return walletContract.instance.m_required.call();
+  }
+
+  static fetchOwners (walletContract) {
+    const walletInstance = walletContract.instance;
+
+    return walletInstance
+      .m_numOwners.call()
+      .then((mNumOwners) => {
+        const promises = range(mNumOwners.toNumber())
+          .map((idx) => walletInstance.getOwner.call({}, [ idx ]));
+
+        return Promise
+          .all(promises)
+          .then((owners) => {
+            const uniqOwners = uniq(owners);
+
+            // If all owners are the zero account : must be Mist wallet contract
+            if (uniqOwners.length === 1 && /^(0x)?0*$/.test(owners[0])) {
+              return WalletsUtils.fetchMistOwners(walletContract, mNumOwners.toNumber());
+            }
+
+            return owners;
+          })
+          .then((owners) => uniq(owners));
+      });
+  }
+
+  static fetchMistOwners (walletContract, mNumOwners) {
+    const walletAddress = walletContract.address;
+
+    return WalletsUtils
+      .getMistOwnersOffset(walletContract)
+      .then((result) => {
+        if (!result || result.offset === -1) {
+          return [];
+        }
+
+        const owners = [ result.address ];
+
+        if (mNumOwners === 1) {
+          return owners;
+        }
+
+        const initOffset = result.offset + 1;
+        let promise = Promise.resolve();
+
+        range(initOffset, initOffset + mNumOwners - 1).forEach((offset) => {
+          promise = promise
+            .then(() => {
+              return walletContract.api.eth.getStorageAt(walletAddress, offset);
+            })
+            .then((result) => {
+              const resultAddress = '0x' + (result || '').slice(-40);
+              const { address } = validateAddress(resultAddress);
+
+              owners.push(address);
+            });
+        });
+
+        return promise.then(() => owners);
+      });
+  }
+
+  static getMistOwnersOffset (walletContract, offset = 3) {
+    return walletContract.api.eth
+      .getStorageAt(walletContract.address, offset)
+      .then((result) => {
+        if (result && !/^(0x)?0*$/.test(result)) {
+          const resultAddress = '0x' + result.slice(-40);
+          const { address, addressError } = validateAddress(resultAddress);
+
+          if (!addressError) {
+            return { offset, address };
+          }
+        }
+
+        if (offset >= 100) {
+          return { offset: -1 };
+        }
+
+        return WalletsUtils.getMistOwnersOffset(walletContract, offset + 1);
+      });
+  }
+
+  static fetchDailylimit (walletContract) {
+    const walletInstance = walletContract.instance;
+
+    return Promise
+      .all([
+        walletInstance.m_dailyLimit.call(),
+        walletInstance.m_spentToday.call(),
+        walletInstance.m_lastDay.call()
```

Comment:
Again, any particular reason for using `call` instead of just calling the method?